### DISCUSSION TITLE
Is there a timeline for this to be released?

### DISCUSSION THREAD

**Discussion by abder** [mdx-js/mdx#2330] [General] [answered]
Hey everyone, I'm curious about the release timeline for the MDX TypeScript support in Visual Studio Code. Does anyone have any information on when we can expect this feature to be available? Your insights would be much appreciated! Thanks in advance.

**Comment by remcohaszing** [comment]
You can already enable editor features by setting [`"mdx.experimentalLanguageServer": true`](https://github.com/mdx-js/mdx-analyzer/tree/main/packages/vscode-mdx#settings). A list of supported features is [available here](https://github.com/mdx-js/mdx-analyzer/blob/main/packages/language-server/README.md#language-server-features).

I’m currently reworking this to work with [Volar](https://volarjs.github.io), which should provide more stability and allows us to ship more features faster.

**Reply by abder** [reply]
@remcohaszing Thanks for the reply 😄 

I do have this enabled actually, I am not sure if I have stated what I am looking for correctly, but what I am looking for is when I set a wrong type for a component prop in MDX it will be highlighted and shows the error, in the same way errors are highlighted in ts and tsx files 

Am I missing something?

**Reply by remcohaszing** [reply, marked as answer]
Under the hood MDX files are treated the same as JavaScript files. This means that by default IntelliSense works, but not strict type checking. To enable that, set `"checkJs": true` in your `tsconfig.json`.

**Comment by abder** [comment]
@remcohaszing That has worked!! was really helpful, thanks 😃
